### ISSUE TITLE
update.py should remove valueFrom under inputBinding in v1.0

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/cwltool#1064]
`cwltool --validate https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/workflows/bamfastq_align/etl.cwl`

works with v1.0.20181217162649 but not with the unreleased 1.0.20190205170641

```
$ cwltool --debug --validate https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/workflows/bamfastq_align/etl.cwl
/home/michael/cwltool/env3/bin/cwltool 1.0.20190205170641
Tool definition failed initialization:
https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/workflows/bamfastq_align/extract_readgroup_fastq_pe.cwl:12:9: Reference `#` not found in file `https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/tools/readgroup.yml`.
Traceback (most recent call last):
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 609, in resolve_ref
    % (frg, doc_url))
schema_salad.validate.ValidationException: Reference `#` not found in file `https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/tools/readgroup.yml`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/main.py", line 675, in main
    tool = make_tool(uri, loadingContext)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/load_tool.py", line 351, in make_tool
    tool = loadingContext.construct_tool_object(processobj, loadingContext)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/workflow.py", line 55, in default_make_tool
    return Workflow(toolpath_object, loadingContext)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/workflow.py", line 535, in __init__
    loadingContext.prov_obj))
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/workflow.py", line 569, in make_workflow_step
    return WorkflowStep(toolpath_object, pos, loadingContext, parentworkflowProv)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/workflow.py", line 630, in __init__
    toolpath_object["run"], loadingContext)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/load_tool.py", line 370, in load_tool
    loadingContext, workflowobj, uri)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/cwltool/load_tool.py", line 292, in resolve_and_validate_document
    workflowobj, fileuri, checklinks=loadingContext.do_validate)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 956, in resolve_all
    val, base_url+subscope, file_base=file_base, checklinks=False)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 990, in resolve_all
    val, base_url, file_base=file_base, checklinks=False)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 956, in resolve_all
    val, base_url+subscope, file_base=file_base, checklinks=False)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 969, in resolve_all
    val, base_url=file_base, checklinks=False)
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/ref_resolver.py", line 610, in resolve_ref
    doc = self.fetch(doc_url, inject_ids=(not mixin))
  File "/home/michael/cwltool/env3/lib/python3.7/site-packages/schema_salad/sourceline.py", line 168, in __exit__
    raise self.makeError(six.text_type(exc_value))
RuntimeError: https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/workflows/bamfastq_align/extract_readgroup_fastq_pe.cwl:12:9: Reference `#` not found in file `https://raw.githubusercontent.com/NCI-GDC/gdc-dnaseq-cwl/017c7572ea309c7d5b34bcc9bc1bdafbe47cb515/tools/readgroup.yml`.
Command exited with non-zero status 1
$ pip show schema_salad
Name: schema-salad
Version: 4.0.20190130225346
```

**Comment by tetron** [comment]
Yea I'm seeing this when trying to generate the web site.  Will fix.

**Comment by tetron** [comment]
Fix: https://github.com/common-workflow-language/cwltool/pull/1069

**Comment by tetron** [comment]
For this specific file, there's an unrelated validation error, there's an ExpressionTool `emit_json_value.cwl` which passes validation on its own, but when validated as part of the full workflow, it gets a validation error.

Here's the fragment in question:

```
inputs:
  - id: input
    type: File
    inputBinding:
      loadContents: true
      valueFrom: $(null)
```

The `valueFrom: $(null)` under `inputBinding` is illegal.  It passes validation in v1.0 but that was an inintended mistake in the schema.

I think what is happening is that on its own it is only validated once, using the v1.0 schema, but for some reason as part of the full workflow, it is being (re)validated as v1.1, which is stricter.

        * `Workflow.inputs` and `ExpressionTool.inputs` now refer to the type
          `WorkflowInputParameter` instead of `InputParameter`; the only syntax
          difference from v1.0 is that `inputBinding` is now deprecated (as
          there is no command line to bind input values to) and `inputBinding`
          in a `Workflow` context will be removed in v2.0. Use the new
          `WorkflowInputParameter.loadContents` instead.


Unlike `loadContents` (which does actually affect behavior, and was implemented),`valueFrom` was never specified to do anything in this context, so making it valid (and specifying behavior) in v1.1 is just as likely to break stuff as the current policy of it being invalid.

**Comment by tetron** [comment]
Just had a thought that maybe the updater could delete the field (with a warning).

**Comment by mr-c** [comment]
> Just had a thought that maybe the updater could delete the field (with a warning).

Agreed

**Comment by mr-c** [comment]
This was fixed in #1071
